products/views.py

```diff
@@ -1,15 +1,17 @@
-from tempfile import _TemporaryFileWrapper
 from django.views import View
-from django.http import JsonResponse
+from django.http  import JsonResponse
 
-from .models import Product, Image
+from .models      import Product
 
-class ProductGroupDetailView(View):
-    def get(self, request, productgroup_id):
-        if not Product.objects.filter(productgroup_id=productgroup_id).exists():
-            return JsonResponse({'message':'productgroup_id error'}, status=400)
-
-        products = Product.objects.filter(productgroup_id=productgroup_id).order_by('is_default')
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        if not Product.objects.filter(id=product_id).exists():
+            return JsonResponse({'message':'product_id error'}, status=400)
+            
+        productgroup_id = Product.objects.get(id=product_id).productgroup_id
+        product         = Product.objects.filter(id=product_id)
+        other_products  = Product.objects.filter(productgroup_id=productgroup_id).exclude(id=product_id)
+        products        = product.union(other_products)
```

Comment:
지연님!
여기서 `product와 같은 그룹에 있는 상품들 같이 반환해주는 건가요?
그렇다면 union은 왜 다시 해주는건가요?

Reply:
같은 그룹에 있는 상품들을 반환해주는 용도입니다! 클릭한 상품이 먼저 나올 수 있도록 나머지 상품들을 union으로 합쳤습니다.

Reply:
반환해주는 데이터 구조를 수정할 필요가 있어 보입니다.
우선 Client에서 product_id = 1인 상품 정보를 요청했으면, 1번 상품이 root query로 반환되어야 합니다. 아래 json 구조를 참고해주세요~
```json
{
    "id" : product.id,
    "name" : product.name,
    "related_products" : [
        {
              "id" : product.id,
              "name" : product.name,
         },
        {
              "id" : product.id,
              "name" : product.name,
         },
    ]
}
```

```suggestion
        product = Product.objects.get(id=product_id)
        related_products  = Product.objects.filter(productgroup_id=productgroup_id).exclude(id=product_id)
        related_product.insert(0, product)
```